### ISSUE TITLE
Compile issues using ES Modules

### ISSUE THREAD

**Issue by t2farrell** [elastic/enterprise-search-js#16]
## 🐛 Bug Report

There appears to be a bug that causes type errors and compilation failure if the @elastic/enterprise-search module is used with ES Modules.

## To Reproduce

Note that in the tsconfig.json file the property "skipLibCheck": false must be set (or the default generated setting of true must be commented out).

Your README example uses `require` and will work fine.
```js
const { Client } = require('@elastic/enterprise-search')
```

However, the following will not compile: 

```js
import { Client } from '@elastic/enterprise-search';
```

Using the above will fail with type errors, such as:

```
node_modules/@elastic/enterprise-search/index.d.ts:22:49 - error TS2307: Cannot find module './lib/types' or its corresponding type declarations.
```

## Expected behavior

A clean compilation without errors.

## Your Environment

- *node version*: 14, 16, 18
- `@elastic/enterprise-search` *version*: ^8.1.0-beta.1
- *os*: Windows, Linux

**Comment by sethmlarson** [comment]
@t2farrell Thanks for opening this. I'm not a JavaScript developer so take my suggestions with a grain of salt. I was able to reproduce the issue you're seeing locally and have created this PR: https://github.com/elastic/enterprise-search-js/pull/17 Can you take a look and see if this matches up with what you're imagining a fix looks like?

**Comment by t2farrell** [comment]
@sethmlarson Thanks for investigating this.  Yes, your changes match the local changes we made to work around the issue.

**Comment by sethmlarson** [comment]
I've released 8.1.1-beta.1 which includes this fix: https://www.npmjs.com/package/@elastic/enterprise-search/v/8.1.1-beta.1

**Comment by t2farrell** [comment]
Thanks!
